plugins/copr.py

```diff
@@ -465,35 +467,32 @@ def _download_repo(self, project_name, repo_filename):
         arch = self.chroot_parts[-1]
         api_path = "/coprs/{0}/repo/{1}/dnf.repo?arch={2}".format(project_name, short_chroot, arch)
 
-        try:
-            f = self.base.urlopen(self.copr_url + api_path, mode='w+')
-        except IOError as e:
-            if os.path.exists(repo_filename):
-                os.remove(repo_filename)
-            if '404' in str(e):
-                if PY3:
-                    import urllib.request
-                    try:
-                        res = urllib.request.urlopen(self.copr_url + "/coprs/" + project_name)
-                        status_code = res.getcode()
-                    except urllib.error.HTTPError as e:
-                        status_code = e.getcode()
-                else:
-                    import urllib
-                    res = urllib.urlopen(self.copr_url + "/coprs/" + project_name)
+        response = requests.get(self.copr_url + api_path)
+        if response.status_code != 200:
+            if PY3:
+                import urllib.request
+                try:
+                    res = urllib.request.urlopen(self.copr_url + "/coprs/" + project_name)
                     status_code = res.getcode()
-                if str(status_code) != '404':
-                    raise dnf.exceptions.Error(_("This repository does not have"
-                                                 " any builds yet so you cannot enable it now."))
-                else:
-                    raise dnf.exceptions.Error(_("Such repository does not exist."))
-            raise
-
-        for line in f:
-            if re.match(r"\[copr:", line):
-                repo_filename = os.path.join(self.base.conf.get_reposdir,
-                                             "_" + line[1:-2] + ".repo")
-            break
+                except urllib.error.HTTPError as e:
+                    status_code = e.getcode()
+            else:
+                import urllib
+                res = urllib.urlopen(self.copr_url + "/coprs/" + project_name)
+                status_code = res.getcode()
+            if str(status_code) != '404':
+                msg = base64.b64decode(response.headers["Error-Message"]).decode('utf-8')
```

Comment:
We should use `response.headers.get()` instead, just in case the Error-Message isn't provided.